File: server/ruby/app/infrastructure/mappers/payment_details/payment_product840_customer_account_mapper.rb
module Infrastructure
  module Mappers
    module PaymentDetails
      class PaymentProduct840CustomerAccountMapper
        def self.from_sdk_response(response)
          return nil if response.nil?

          dto = Business::Domain::Payments::PaymentDetails::PaymentProduct840CustomerAccount.new
          dto.account_id = response.account_id
          dto.company_name = response.company_name
          dto.country_code = response.country_code
          dto.first_name = response.first_name
          dto.customer_account_status = response.customer_account_status
          dto.customer_address_status = response.customer_address_status
          dto.payer_id = response.payer_id
          dto.surname = response.surname

          dto
        end
      end
    end
  end
end

File: server/nodejs/src/infrastructure/mappers/payment-details/payment-product-840-customer-account-mapper.ts
import type { Domain } from 'onlinepayments-sdk-nodejs';
import { PaymentProduct840CustomerAccount } from '../../../business/domain/payments/payment-details/payment-product-840-customer-account';

type PaymentProduct840CustomerAccountSdk = Domain.PaymentProduct840CustomerAccount;

export const PaymentProduct840CustomerAccountMapper = {
    fromSdkResponse: (
        response?: PaymentProduct840CustomerAccountSdk | null,
    ): PaymentProduct840CustomerAccount | undefined => {
        if (!response) {
            return undefined;
        }

        return {
            accountId: response?.accountId,
            companyName: response?.companyName,
            countryCode: response?.countryCode,
            firstName: response?.firstName,
            customerAccountStatus: response?.customerAccountStatus,
            customerAddressStatus: response?.customerAddressStatus,
            payerId: response?.payerId,
            surname: response?.surname,
        };
    },
};
